splink/athena/athena_linker.py

```diff
@@ -67,7 +67,7 @@ def columns(self):
     def validate(self):
         pass
 
-    def drop_table_from_database(self, force_non_splink_table=False):
+    def _drop_table_from_database(self, force_non_splink_table=False):
```

Comment:
This is renamed because we want end-users to use `drop_table_from_database_and_remove_from_cache`.  

We don' want them to us `_drop_table_from_database` on its own because this would leave the table in the cache, which would cause errors when the table was assumed to exist but had been dropeed